compiler/rustc_parse/src/errors.rs

```diff
@@ -2636,21 +2636,13 @@ pub(crate) struct MissingPlusBounds {
 }
 
 #[derive(Diagnostic)]
-#[diag(parse_incorrect_braces_trait_bounds)]
-pub(crate) struct IncorrectBracesTraitBounds {
+#[diag(parse_incorrect_parens_trait_bounds)]
+pub(crate) struct IncorrectParensTraitBoundsSugg {
     #[primary_span]
-    pub span: Vec<Span>,
-    #[subdiagnostic]
-    pub sugg: IncorrectBracesTraitBoundsSugg,
-}
-
-#[derive(Subdiagnostic)]
-#[multipart_suggestion(parse_suggestion, applicability = "machine-applicable")]
-pub(crate) struct IncorrectBracesTraitBoundsSugg {
-    #[suggestion_part(code = " ")]
-    pub l: Span,
-    #[suggestion_part(code = "")]
-    pub r: Span,
```

Comment:
Could we keep the multipart suggestion approach? Even if we suggest moving the brace to the left of the `dyn`, the "patch-style" suggestion format tends to be easier to understand for people.

Reply:
yes, fixed it.